File: frontend/shared/services/searchService.ts
import { apiClient } from "./apiClient";

export interface SearchHit {
  documentId: string;
  versionId: string;
  title: string;
  filename: string;
  score: number;
  snippet?: string;
  highlights?: string[];
  pageReferences?: number[];
}

export interface SearchResult {
  query: string;
  searchType: string;
  totalHits: number;
  tookMs: number;
  hits: SearchHit[];
}

export const searchService = {
  async searchLexicalOrHybrid(query: string, searchType: string = "hybrid", limit: number = 10): Promise<SearchResult> {
    const response = await apiClient.get("/search", {
      params: { query, searchType, limit },
    });
    return response.data;
  },

  async searchSemantic(query: string, limit: number = 10): Promise<SearchResult> {
    const response = await apiClient.post("/semantic-search", {
      query,
      limit,
    });
    return response.data;
  },
};

File: backend/src/main/java/com/enterprise/platform/modules/search/service/SearchService.java
package com.enterprise.platform.modules.search.service;

import com.enterprise.platform.modules.search.provider.SearchResult;

public interface SearchService {
    SearchResult search(String query, String tenantId, String permissionHash, int limit) throws Exception;
    SearchResult search(String query, String tenantId, String permissionHash, String searchType, int limit) throws Exception;
}
